packages/block-editor/src/components/inner-blocks/test/get-block-context.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	getBlockType,
+	getBlockTypes,
+	registerBlockType,
+	unregisterBlockType,
+} from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import getBlockContext from '../get-block-context';
+
+describe( 'getBlockContext', () => {
+	beforeAll( () => {
+		registerBlockType( 'core/fruit-basket', {
+			category: 'text',
+			title: 'Fruit Basket',
+			attributes: {
+				id: {
+					type: 'number',
+				},
+				name: {
+					type: 'string',
+				},
+			},
+			providesContext: { basketId: 'id' },
+		} );
+	} );
+	afterAll( () => {
+		getBlockTypes().forEach( ( block ) => {
+			unregisterBlockType( block.name );
+		} );
+	} );
+
+	it( 'should return cached value when attributes are the same', () => {
+		const attributes = { id: 1, name: '' };
+		const blockType = getBlockType( 'core/fruit-basket' );
+
+		const prevContext = getBlockContext( attributes, blockType );
+		const nextContext = getBlockContext( attributes, blockType );
+
+		expect( prevContext ).toBe( nextContext );
+	} );
+
+	it( 'should return cached value if attributes used in context are the same', () => {
```

Comment:
Should we also add a test for the case when there's no `providesContext`?

Reply:
The `blockType` object is a required argument, so I'm not sure how helpful that test would be.